feature-flags-tmp/feature_flags/feature_flags.py

```diff
@@ -0,0 +1,222 @@
+        for k in self.pending_deletes:
+            if k in result:
+                result.remove(k)
+        return result
+
+    @rx.var
+    def pending_updates(self) -> set[str]:
+        result = set()
+        for k in self.feature_flags_as_loaded_from_db or {}:
+            if k in self.pending_creates_or_updates:
+                result.add(k)
+        for k in self.pending_deletes:
+            if k in result:
+                result.remove(k)
+        return result
+
+    def update_feature_flag(self, k: str, val: str):
+        self.pending_creates_or_updates[k] = val
+
+    def save_to_db(self):
+        with rx.session() as session:
+            for k, v in self.pending_creates_or_updates.items():
+                if k not in self.pending_deletes:
+                    # TODO: Do we have to select the item before updating it?
+                    ff = session.exec(FeatureFlags.select.where(FeatureFlags.name == k)).first()
+                    if not ff:
+                        ff = FeatureFlags(name=k, value=v)
+                    else:
+                        ff.value = v
+                    session.add(ff)
+
+            for k in self.pending_deletes:
+                # TODO: Do we have to select the item before deleting it?
+                ff = session.exec(FeatureFlags.select.where(FeatureFlags.name == k)).first()
+                if ff:
+                    session.delete(ff)
+            session.commit()
+        self.pending_deletes = set()
+        self.pending_creates_or_updates = {}
+        self.load_feature_flags_from_db()
+
+    def reset_new_flag_modal_state(self):
+        self.new_flag_modal_is_open = False
+        self.new_flag_modal_error = None
+        self.new_flag_modal_flag_name = ""
+        self.new_flag_modal_flag_value = ""
+
+    def new_flag_modal_stage(self):
+        if re.search(r"[^a-zA-Z0-9_]+", self.new_flag_modal_flag_name):
+            self.new_flag_modal_error = "Flag name must be contain only one or more alphanumeric or _ chars"
+            return
+        if self.new_flag_modal_flag_name in self.latest_feature_flags_view():
+            self.new_flag_modal_error = "Flag already exists"
+            return
+        self.pending_creates_or_updates[self.new_flag_modal_flag_name] = self.new_flag_modal_flag_value
+        self.reset_new_flag_modal_state()
+
+    def new_flag_modal_cancel(self):
+        self.reset_new_flag_modal_state()
+
+    def stage_delete_feature_flag(self, k: str):
+        self.pending_deletes.add(k)
+
+
+def index() -> rx.Component:
+    return rx.fragment(
+        rx.color_mode_button(rx.color_mode_icon(), float="right"),
+        rx.vstack(
+            rx.heading("Flex-Flags", font_size="2em"),
+            rx.hstack(
+                rx.button("Save", on_click=FeatureFlagsState.save_to_db, color=FeatureFlagsState.save_button_color),
+                rx.button("Add new", on_click=lambda: FeatureFlagsState.set_new_flag_modal_is_open(True)),
```

Comment:
lambda not actually needed unless you're passing through parameters

```suggestion
                rx.button("Add new", on_click=FeatureFlagsState.set_new_flag_modal_is_open(True)),
```

Reply:
Ok good to know.

IMO it's quite unexpected.  I'm confused by exactly the type of `on_click`.  Is it a callable?  Or is it the result of a callable?

Reply:
the on_click ultimately wants an `EventSpec`, which is basically an `EventHandler` with concrete arguments. If you pass a lambda, we call it internally and it must return an `EventSpec`. Calling an `EventHandler` returns an `EventSpec`